packages/dds/task-manager/src/taskManager.ts

```diff
@@ -254,18 +259,18 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
         this.latestPendingOps.set(taskId, pendingOp);
     }
 
-    public async lockTask(taskId: string) {
+    public async volunteerForTask(taskId: string) {
         // If we have the lock, resolve immediately
-        if (this.haveTaskLock(taskId)) {
-            return;
+        if (this.assigned(taskId)) {
+            return true;
         }
 
         if (!this.connected) {
             throw new Error(`Attempted to lock in disconnected state: ${taskId}`);
         }
 
         // This promise works even if we already have an outstanding volunteer op.
-        const lockAcquireP = new Promise<void>((resolve, reject) => {
+        const lockAcquireP = new Promise<boolean>((resolve, reject) => {
```

Comment:
I think the purpose of switching from a Promise<void> to Promise<boolean> is in anticipation of adding the "completed" op.  Probably would have been better to make the switch at the same time as adding that feature (since we don't gain anything by making the switch until we have a reason to resolve(false)), but OK to leave since I think you'll be following up with that change shortly.

Reply:
Yeah, I probably should have just waited for this change, but I agree that it's probably fine in this PR since the completed stuff will be in my next PR.